zkevm-circuits/src/evm_circuit/execution/returndatasize.rs

```diff
@@ -0,0 +1,186 @@
+            ),
+        )?;
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::evm_circuit::test::rand_bytes;
+    use crate::test_util::run_test_circuits;
+    use eth_types::{bytecode, ToWord, Word};
+    use mock::test_ctx::TestContext;
+
+    fn test_ok_internal(return_data_offset: usize, return_data_size: usize) {
+        let (addr_a, addr_b) = (mock::MOCK_ACCOUNTS[0], mock::MOCK_ACCOUNTS[1]);
+
+        let pushdata = rand_bytes(32);
+        let code_b = bytecode! {
+            PUSH32(Word::from_big_endian(&pushdata))
+            PUSH1(0)
+            MSTORE
+
+            PUSH32(return_data_size)
+            PUSH1(return_data_offset)
+            RETURN
+            STOP
+        };
+
+        // code A calls code B.
+        let code_a = bytecode! {
+            // call ADDR_B.
+            PUSH32(return_data_size) // retLength
+            PUSH1(return_data_offset) // retOffset
+            PUSH1(0x00) // argsLength
+            PUSH1(0x00) // argsOffset
+            PUSH1(0x00) // value
+            PUSH32(addr_b.to_word()) // addr
+            PUSH32(0x1_0000) // gas
+            CALL
+            RETURNDATASIZE
+            STOP
+        };
+
+        let ctx = TestContext::<3, 1>::new(
+            None,
+            |accs| {
+                accs[0].address(addr_b).code(code_b);
+                accs[1].address(addr_a).code(code_a);
+                accs[2]
+                    .address(mock::MOCK_ACCOUNTS[2])
+                    .balance(Word::from(1u64 << 30));
+            },
+            |mut txs, accs| {
+                txs[0].to(accs[1].address).from(accs[2].address);
+            },
+            |block, _tx| block,
+        )
+        .unwrap();
+
+        assert_eq!(run_test_circuits(ctx, None), Ok(()));
+    }
+
+    #[test]
+    fn returndatasize_gadget_simple() {
+        test_ok_internal(0x00, 0x02);
+    }
+
+    #[test]
+    fn returndatasize_gadget_large() {
+        test_ok_internal(0x00, 0x20);
+    }
+
+    #[test]
+    #[should_panic]
+    fn returndatasize_gadget_out_of_bound() {
+        test_ok_internal(0x00, 0x0101);
+    }
+
+    #[test]
+    fn returndatasize_gadget_zero_length() {
+        test_ok_internal(0x00, 0x00);
+    }
+
+    #[test]
+    fn test_simple() {
+        let (addr_a, addr_b) = (mock::MOCK_ACCOUNTS[0], mock::MOCK_ACCOUNTS[1]);
+        let code = bytecode! {
+            RETURNDATASIZE
+            STOP
+        };
+        let ctx = TestContext::<2, 1>::new(
+            None,
+            |accs| {
+                accs[0].address(addr_b).code(code);
+                accs[1].address(addr_a).balance(Word::from(1u64 << 30));
+            },
+            |mut txs, accs| {
+                txs[0].to(accs[0].address).from(accs[1].address);
+            },
+            |block, _tx| block,
+        )
+        .unwrap();
+
+        assert_eq!(run_test_circuits(ctx, None), Ok(()));
+    }
```

Comment:
Not a biggie, but I think you can simplify this right? Or am I missing something?

```suggestion
    assert_eq!(
        run_test_circuits(
            TestContext::<2, 1>::simple_ctx_with_bytecode(code).unwrap(),
            None              
        ),                                                                                                                 
        Ok(())
    );
```

Reply:
Updated. Thanks.